### PR TITLE
Make generation of the apply annotation optional

### LINKED ISSUE #16800: Make generation of the "apply annotation" optional
Add a flag to allow the users to specify if they want the "apply annotation".

### PR DESCRIPTION
**Pull request by janetkuo**
Fixes #16800

Add a boolean flag `--save-config` (default `false`) to `kubectl create`, `edit`, `run`, `expose`, `replace`, and `autoscale`. If the flag is set to `true`, create the apply configuration annotation no matter what; otherwise, update the annotation only when it already exists. 

cc @kubernetes/kubectl

### CODE REVIEW

**Comment by bgrant0607** [comment]
cc @jackgr @jlowdermilk

**Comment by bgrant0607** [comment]
I'm ok with `--save-config`. Assigning to @jackgr for the rest of the review.

cc @liggitt

**Comment by jackgr** [comment]
Looks good, but needs tests.

**Comment by janetkuo** [comment]
Tests added. I found that the update annotation function doesn't work in kubectl edit. Need to fix this.

**Comment by jackgr** [comment]
What's the problem?

File: pkg/kubectl/cmd/edit.go [hunk 1]
```diff
@@ -220,8 +221,7 @@ func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 				return fmt.Errorf("The edited file had a syntax error: %v", err)
 			}
 
-			// annotate the edited object for kubectl apply
-			if err := kubectl.UpdateApplyAnnotation(updates); err != nil {
+			if err := kubectl.CreateOrUpdateAnnotation(cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag), updates); err != nil {
```
Comment (janetkuo) [inline pkg/kubectl/cmd/edit.go]: @jackgr after updating annotations of "updates" (`Info` of the edited file), the config annotation is added to it successfully. However, as you can see, only its data, "edited", will be used to generate a "patch" (see https://github.com/janetkuo/kubernetes/blob/apply-annotation-optional/pkg/kubectl/cmd/edit.go#L244). So the "patch" won't contain the config annotation we added earlier while we do Patch(), see https://github.com/janetkuo/kubernetes/blob/apply-annotation-optional/pkg/kubectl/cmd/edit.go#L264.

Comment (jackgr) [inline pkg/kubectl/cmd/edit.go]: Thanks, @janetkuo.

**Comment by jackgr** [comment]
LGTM

**Comment by janetkuo** [comment]
@jackgr I fixed the kubectl edit problem in the second commit. PTAL.

**Comment by jackgr** [comment]
LGTM

**Comment by bgrant0607** [comment]
Tests failed

**Comment by janetkuo** [comment]
Squashed commits

**Pull request merged**